float64 target
float64 quad_pos